src/arch/ppc64.cpp

```diff
@@ -99,6 +104,15 @@ int sp_offset()
   return offset("r1");
 }
 
+int arg_stack_offset()
+{
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+  return ARG0_STACK_LE / 8;
+#else
+  return ARG0_STACK_BE / 8;
+#endif
```

Comment:
So @dalehamel added support for static builds of bpftrace that are built on every commit / release. And @fbs added better support for runtime detection of kernel features.

Perhaps this should be a runtime check as well (ie `is_little_endian()`)? So that we can eventually build a single static bpftrace binary that can run on any linux host. Thoughts @dalehamel @fbs ?

Reply:
I don't think there are any architectures that mix endianness? Linux is either big or little on an architecture (even if the hardware supports both) right? If that won't change there is no reason to detect it at runtime.

Also a single binary that runs on any platform won't work due to the different CPU architectures.

Reply:
> Also a single binary that runs on any platform won't work due to the different CPU architectures.

Yeah right now you can't. But if we wanted, we could have a fatter binary that switches between the different CPU-specific stuff at runtime, right?

Reply:
What I mean is that code (e.g. our static binary) compiled for x86 won't run on ARM.  So LE/BE runtime detection is only useful if there is a platform on which Linux runs as both LE and BE, which I don't think there is.

Reply:
Ah yeah I'm going crazy.